### PR TITLE
qa-tests: Add tests for realworld pattern-regex

### LINKED ISSUE #785: Add regex pattern tests in the QA test suite
The real world's shown that there's a bunch of issues with regex matching: https://github.com/returntocorp/semgrep/issues/781

We should add a regex pattern to `tests/qa/test_public_repos.py` which would run the regex pattern module on lots of real world code.

### PR DESCRIPTION
**Pull request by brendongo**
Closes #785

### CODE REVIEW

File: semgrep/tests/qa/rules/regex-sentinel.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+rules:
+  - id: test-regex
+    patterns:
+      - pattern-regex: '. = 87518275812375164'
```
Comment (underyx) [inline semgrep/tests/qa/rules/regex-sentinel.yaml:4]: aww my lil number working his way into everyone's hearts 😊

File: semgrep/tests/qa/test_public_repos.py [hunk 2]
```diff
@@ -109,3 +114,19 @@ def test_semgrep_on_repo(run_semgrep_in_tmp, repo_url):
         assert output["errors"] == []
         assert len(output["results"]) == 1
         assert output["results"][0]["path"] == str(sentinel_path)
+
+    output = subprocess.check_output(
+        [
+            "python3",
+            "-m",
+            "semgrep",
+            "--config=rules/regex-sentinel.yaml",
```
Comment (underyx) [inline semgrep/tests/qa/test_public_repos.py:123]:
```suggestion
            "--config",
            Path(TESTS_PATH / "qa" / "rules" / "regex-sentinel.yaml").resolve(),
```

wouldn't this work? then we wouldn't need to mess with tmp paths above.

Comment (DrewDennison) [reply semgrep/tests/qa/test_public_repos.py:123]: I think you need to wrap with `str(...)`

Comment (underyx) [reply semgrep/tests/qa/test_public_repos.py:123]: nope, subprocess functions accept `Path`

Comment (underyx) [reply semgrep/tests/qa/test_public_repos.py:123]: Ah, now I see that most of the tmp_path lines would need to stay anyway cause we got rid of the `run_semgrep_in_tmp` fixture which would normally set it up.

File: semgrep/tests/qa/test_public_repos.py [hunk 3]
```diff
@@ -60,7 +60,12 @@ def xfail_repo(url, *, reason=None):
         ),
     ],
 )
-def test_semgrep_on_repo(run_semgrep_in_tmp, repo_url):
+def test_semgrep_on_repo(monkeypatch, tmp_path, repo_url):
+    TESTS_PATH = Path(__file__).parent.parent
```
Comment (mschwager) [inline semgrep/tests/qa/test_public_repos.py:64]: Not really sure what's going on with all this path stuff - I trust @underyx's thoughts :+1:

Comment (underyx) [reply semgrep/tests/qa/test_public_repos.py:64]: it's just replicating the code from `run_semgrep_in_tmp` fixture, we good 😇

**Review by underyx — APPROVED** [review]

**Pull request merged**